apps/website/blog/2023-10-11-frontend-plugin/index.mdx

```diff
@@ -0,0 +1,451 @@
+  - `CpuData`: Type to contain CPU data from the backend plugin JSON response
+  - `SysInfoMainData`: Type to contain data element from the backend plugin JSON response
+  - `SysInfoData`: Type to contain `SysInfoMainData` and list (Array) of `CpuData` data
+  - `sysInfoCpuColumns`: This is a list of fields that are of type `TableColumn` and are used to display the header for CPU columns.
+  - `sysInfoMainDataColumns`: This is a list of fields that are of type `TableColumn` and are used to display the header for main data columns.
+    - _The `field` property for each element in the list maps to the field that is used to display data from the object containing data for the table._
+- **plugins/system-info/src/components/SystemInfoPage/SystemInfoPage.tsx**: Main file to invoke the backend API and parse/display data on the UI with the main lines highlighted in the code block:
+  - Gets the backend baseUrl (`backend.baseUrl`) from the config. This property is automatically configured and is available to the frontend plugin with the use of `configApiRef`
+    ```tsx
+        const config = useApi(configApiRef);
+        ...
+        const backendUrl = config.getString(SYS_INFO_BACKEND_URL);
+    ```
+  - Invokes the backend plugin API (`${backendUrl}/api/sys-info/system-info`) and extracts json from the response as `SysInfoData`
+    ```tsx showLineNumbers {2,3}
+    const backendUrl = config.getString(SYS_INFO_BACKEND_URL);
+    const backendApiEndPoint = `${backendUrl}/api/sys-info/system-info`;
+    const systemInfoData = await fetch(backendApiEndPoint)
+      .then((res) => (res.ok ? res : Promise.reject(res)))
+      .then((res) => res.json());
+    ```
+  - To display the system information data, we are using the following two tables:
+    - First table uses `sysInfoMainDataColumns` for the columns and `sysInfoData?.mainDataAsArray` for the main data (_`mainDataAsArray` is set after fetching the data from backend API_).
+      ```tsx showLineNumbers {8,10}
+          systemInfoData.mainDataAsArray = [];
+          systemInfoData.mainDataAsArray[0] = systemInfoData.data;
+          systemInfoData.mainDataAsArray[0].cpuModel = systemInfoData.cpus[0].model;
+          systemInfoData.mainDataAsArray[0].cpuSpeed = systemInfoData.cpus[0].speed;
+          ...
+          <Table
+            title="System Info Details"
+            columns={sysInfoMainDataColumns}
+            isLoading={isSysInfoLoading}
+            data={sysInfoData?.mainDataAsArray || []}
+      ```
+    - Second table uses `sysInfoCpuColumns` for the columns and `sysInfoData?.cpus` for the CPU data.
+      ```tsx showLineNumbers {3,5}
+          <Table
+            title="System Info Details - CPUs"
+            columns={sysInfoCpuColumns}
+            isLoading={isSysInfoLoading}
+            data={sysInfoData?.cpus || []}
+      ```
+  - Exports the `SystemInfoPage` component, which contains the Grid containing the above two tables components:
```

Comment:
```suggestion
  - Exports the `SystemInfoPage` component, which contains the Grid containing the two table components defined earlier:
```

Reply:
Sorry I realized my older suggestion didn't have the best wording

Reply:
Np. You've done an amazing job of finding every single issue and making this blog a lot better (than what it was when I created the PR) :-)

Reply:
Fixing it now